### DISCUSSION TITLE
Data for time reproduction task (button press duration)

### DISCUSSION THREAD

**Discussion by KarlJXie** [jspsych/jsPsych#2446] [Q&A]
Hi! I'm new to jsPsych, I am creating a task in which the participant is asked to press and hold the button to reproduce the time that previous stimuli displayed. I may missed it, but I do not find a specific plugin which can record this kind of data. I would like to ask if it is possible to use jsPsych to conduct the time reproduction task and collect the data about duration of button press. If yes, how I can achieve it? Please could anyone help me with this? It will be much appreciated!

**Comment by jodeleeuw** [comment]
Hi @KarlJXie,

Are you hoping to measure a keyboard button or a button on the screen?

In either case you'll need to modify one of the existing plugins to record this information, but we can point you in the right direction based on which method you want to use.

**Reply by KarlJXie** [reply]
Hi Josh, Thank you very much for your reply and clarification! In my case, I will measure the duration of a keyboard button press. Specifically, following the target stimuli, the participants will need to press and hold the Spacebar to reproduce the time duration of last stimuli when they see a fixation cross. I hope the software could help me to record the duration of this response. Or it could provide me the timing of the response when the Spacebar is pressed and released, then I will be able to calculate the duration. Thank you!
